### PR TITLE
-checkaction=context: Mark context pointer of structs

### PR DESCRIPTION
**Pull request by MoonlightSentinel**
Alternative to omission implemented in #2720

### CODE REVIEW

**Review by PetarKirov — APPROVED** [review]

File: test/exceptions/src/assert_fail.d [hunk 1]
```diff
@@ -338,6 +338,26 @@ void testStructEquals6()
     test(t1, t3, "T(false, 0, 0, \"\", [], `null`) != T(false, 0, 0, \"\", [], C())");
 }
 
+void testStructOpEquals()
+{
+    struct T1 {
```
Comment (PetarKirov) [inline test/exceptions/src/assert_fail.d:343]:
```suggestion
    struct T1
    {
```

File: test/exceptions/src/assert_fail.d [hunk 2]
```diff
@@ -338,6 +338,26 @@ void testStructEquals6()
...
+        bool opEquals(T1) { return false; }
+    }
+
+    T1 t1;
+    test(t1, t1, "T1() != T1()");
+}
+
+void testStructOpEquals2()
+{
+    struct T1 {
```
Comment (PetarKirov) [inline test/exceptions/src/assert_fail.d:353]:
```suggestion
    struct T1
    {
```

File: test/exceptions/src/assert_fail.d (hunk 1 above)
Comment (PetarKirov) [inline test/exceptions/src/assert_fail.d:343]:
```suggestion
    struct T1
    {
```

File: test/exceptions/src/assert_fail.d (hunk 2 above)
Comment (PetarKirov) [inline test/exceptions/src/assert_fail.d:353]:
```suggestion
    struct T1
    {
```

**Comment by Geod24** [comment]
Can it happen that the struct equality fails only because of different ctx pointer ? In which case the error message would be quite confusing ?

**Comment by MoonlightSentinel** [comment]
Actually yes, so it must not be omitted. 
Maybe this should add a marker for  the context ptr, e.g.
```d
S(2, 0x20714600010 <context>)
S(2, context: 0x20714600010)
???
```

**Comment by Geod24** [comment]
The former sounds better as a struct could have a `context` member.
Alternatively, detecting the difference in stack pointer and printing it if it exists:
```
S(2) != S(3) // Different content, same ctx
Context pointer differ: S(2, <context>: 0x20714600010) != S(3, <context>: 0x20714600010) // Different or same content
```

**Comment by MoonlightSentinel** [comment]
Additional messages would only work well for structs at the top level. Adding a prefix also works for embedded structs (e.g. `S[4]`).

**Review by Geod24 — APPROVED** [review]
LGTM

**Pull request merged**